frontend/src/Task.js

```diff
@@ -1,22 +1,135 @@
-import { FaTrashAlt } from "react-icons/fa";
+import { useAtom } from "jotai";
+import { useEffect, useRef, useState } from "react";
+import { createPortal } from "react-dom";
+import { FaCaretDown } from "react-icons/fa";
+import { useUpdateTask } from "./api/api.hooks";
+import { openMenuAtom } from "./atoms";
 
-import { Button } from "./Button";
+const badgeStyle = {
+  backgroundColor: "black",
+  color: "white",
+  display: "inline-block",
+  textAlign: "center",
+  width: "20px",
+  height: "20px",
+  marginRight: "10px",
+  fontSize: "10px",
+  borderRadius: "25%",
+};
 
 export const Task = ({ task, deleteTask }) => {
+  const MenuButton = () => {
+    const [openMenu, setOpenMenu] = useAtom(openMenuAtom);
+    const openMenuButton = useRef(null);
+
+    const menuButtonStyle = {
+      display: "flex",
+      backgroundColor: "darkslategrey",
+      alignItems: "center",
+      justifyContent: "center",
+      cursor: "pointer", // reference: https://stackoverflow.com/questions/3087975/how-to-change-the-cursor-into-a-hand-when-a-user-hovers-over-a-list-item
+      borderRadius: "8px",
+    };
+
+    const openMenuButtonStyle = {
+      paddingRight: "15px",
+      position: "relative",
+    };
+
+    const deleteItemButtonStyle = {
+      paddingLeft: "15px",
+      paddingRight: "15px",
+    };
+
+    const superMenu = {
+      position: "fixed",
+      backgroundColor: "white",
+      padding: "15px",
+      borderRadius: "8px",
```

Comment:
Seems like the default `padding`, when used, is `15px` as well as the default `border radius`, when used, is `8px`. Can we create specific classes for them and apply directly into the components? Then we can define here just the specific styles.